engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -417,11 +417,25 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
       String appPath = response.getString("appPath");
       String user = response.getString("user");
       JSONObject permissionsInfo = null;
-      try (FSDataInputStream is = fileSystem.open(new Path(appPath, "permissions.json"))) {
-        permissionsInfo = new JSONObject(IOUtils.toString(is));
+      Path permissionsPath = new Path(appPath, "permissions.json");
+      LOG.debug("Checking for permission information in file {}", permissionsPath);
+      try {
+        if (fileSystem.exists(permissionsPath)) {
+          LOG.info("Loading permission information");
+          try (FSDataInputStream is = fileSystem.open(permissionsPath)) {
+            permissionsInfo = new JSONObject(IOUtils.toString(is));
+          }
+        } else {
+          // ignore and log messages if file is not found
+          LOG.info("Permission information not available, file not present");
+        }
       } catch (IOException ex) {
-        // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        // ignore and log message when unable to read the file
+        if (LOG.isDebugEnabled()) {
```

Comment:
It would be every time the application info is fetched for any application. Logging only once would require to keep state per application as different applications could be under different paths.

StramAgent is probably not the best place but I think PermissionInfo isn't the best place either as loading from HDFS is not a standard thing that it should be part of it IMO. I would prefer it be handled by a separate loader. I would like to defer that to a later time when we can do a more in-depth look of this feature.

As you mentioned earlier you are ok with merging the PR without refactoring and I would like to defer the refactoring, could you please go ahead with the merge.